gravity.sh

```diff
@@ -591,8 +592,40 @@ gravity_DownloadBlocklistFromUrl() {
     fi
   fi
 
-  # shellcheck disable=SC2086
-  httpCode=$(curl --connect-timeout ${curl_connect_timeout} -s -L ${compression} ${cmd_ext} ${heisenbergCompensator} -w "%{http_code}" "${url}" -o "${listCurlBuffer}" 2>/dev/null)
+  # If we are going to "download" a local file, we first check if the target
+  # file has a+r permission. We explicitly check for all+read because we want
+  # to make sure that the file is readable by everyone and not just the user
+  # running the script.
+  if [[ $url == "file://"* ]]; then
+    # Get the file path
+    file_path=$(echo "$url" | cut -d'/' -f3-)
+    # Check if the file exists and is a regular file (or a symlink to one)
+    if [[ ! -e $file_path ]]; then
+      # Output that the file does not exist
+      echo -e "${OVER}  ${CROSS} ${file_path} does not exist"
+      download=false
+    elif [[ ! -f $file_path ]]; then
+      # Output that the file is not a regular file
+      echo -e "${OVER}  ${CROSS} ${file_path} is not a regular file"
+      download=false
+    else
+      # Check if the file has a+r permissions
+      permissions=$(stat -Lc "%a" "$file_path")
+      if [[ $permissions == *4 || $permissions == *5 || $permissions == *6 || $permissions == *7 ]]; then
```

Comment:
Do we need to check for `5`|`7` since those are directory or exectuable bits? 

Would it be good to expand on the message at 619 to say more about why it's not being used? Something that would say "Executable bit set ${permissions}, ${file_path} not used."

Reply:
Yes, we need to check. Everything that has 4 is readable alone (`4--`). 5 is 4+1 (`r-x`), 6 is 4+2 (`rw-`), 7 is 4+2+1 (`rwx`). We need to check for all of them to get all cases where the readable bit is set.

I did not exclude the cases where `x` is set as there are legit filesystems lacking the concept of proper file permissions. Take, for instance, FAT32 which does not support file permissions. The permissions for everything on a FAT32 formatted device are determined by how the drive is mounted - the default is `0755` so `x` set for everybody. I don't think we should exclude files on FAT32 (also original FAT and even the newer exFAT) by such a rule.